### ISSUE TITLE
java.lang.UnsupportedOperationException: Manual close is not allowed over a Spring managed SqlSession

### ISSUE THREAD

**Issue by mallim** [mybatis/mybatis-spring-boot#9]
The following warning message is observe upon shutting down:

````
2015-12-01 13:57:04.115  WARN 17396 --- [       Thread-4] o.s.b.f.support.DisposableBeanAdapter    : Invocation of destroy method 'close' failed on bean with name 'sqlSessionTemplate': java.lang.UnsupportedOperationException: Manual close is not allowed over a Spring managed SqlSession
13:57:04 WARN  o.s.b.f.s.DisposableBeanAdapter - Invocation of destroy method 'close' failed on bean with name 'sqlSessionTemplate': java.lang.UnsupportedOperationException: Manual close is not allowed over a Spring managed SqlSession
````

**Comment by s17t** [comment]
Confirm this.

**Comment by mallim** [comment]
@s17t  do you mean you also have the same problem? Or will let me to provide more details to confirm this ...

**Comment by mallim** [comment]
After further digging, I just realised that there is an open issue in mybatis/spring#77. Is it possible to have a flag, to indicate whether we want to create SqlSessionTemplate or not? E.g.

````
mybatis.createSqlSessionTemplate = false
````

**Comment by s17t** [comment]
I have this little issue too.

**Comment by mmaquevice** [comment]
+1

**Comment by lpearson05** [comment]
I noticed this issue after bumping the spring version from `4.2.3` to `4.2.4`. I've bumped it back down for the time being.

**Comment by eddumelendez** [comment]
Fixed 990b3e3b123af87303fe6a588a0ddd6a36178bc2
